Final updates to fred factors PCA for code review

arorap95 opened the pull request.

gombrowicz94 commented:
Great work! Just a few comments:


- removing outliers: In terms of removing outliers, would it be better to have the if statement in the actual top-line-function? For a bit I was confused if they were removed regardless, but now I see the if statement is in the remove_outliers function. On this subject - should the default be not to remove outliers? If someone just uses this quickly they may not realize that there is some sort of data change happening.
- factor selection: in the commenting, would it be possible to specify that if you don't specify factor selection, you just get all factors?
- Very nitpicky but can we change any camel case in the functions to standard Python function format? E.g., get_fred_factors() 
- Missing data - this is actually somewhat of a thorny issue for FRED-MD and QD because some series begin later in the series. Perhaps there should be a missing data warning of sorts?

arorap95 commented:
> Great work! Just a few comments:
> 
> * `_removeoutliers`: In terms of removing outliers, would it be better to have the if statement in the actual top-line-function? For a bit I was confused if they were removed regardless, but now I see the if statement is in the `_removeoutliers` function. On this subject - should the default be not to remove outliers? If someone just uses this quickly they may not realize that there is some sort of data change happening.
> * factor selection: in the commenting, would it be possible to specify that if you don't specify factor selection, you just get all factors?
> * Very nitpicky but can we change any camel case in the functions to standard Python function format? E.g., `get_fred_factors() `
> * Should there be type annotations in the functions of the class? I see in the `__init__` call but not elsewhere
> * Missing data - this is actually somewhat of a thorny issue for FRED-MD and QD because some series begin later in the series. Perhaps there should be a missing data warning of sorts?

Thanks so much!
I have made the following changes
-remove outliers default is now False
- changed functions to camel case
-  added type annotations to all functions
- added a warning on missing data
- commented on the factor selection, tat if you don't specify, it defaults to max factors (which defaults to all factors)
Kindly approve if you are good now!

gombrowicz94 commented:
Awesome, thanks! Approving.
